### ISSUE TITLE
MAC should be used to protect sensitive and other files

### ISSUE THREAD

**Issue by liilac** [loki-project/loki-network#111]
See #17 

> A related issue here is using Mandatory Access Control to protect these and other sensitive files.
> 
> macOS and Linux have strong support for this, and offer far more than what file permissions could provide. The situation is more dubious with BSDs as e.g. TrustedBSD is a thing, but there isn't much in the way of actual use of it; it'd be something users enable specifically for Loki.
> 
> If MAC support is of interest, I'm quite comfortable writing and testing policies.

@KeeJef thoughts?

**Comment by neuroscr** [comment]
why a separate issue if we have #17?

**Comment by liilac** [comment]
@neuroscr Mandatory Access Control is separate to (discretionary) filesystem permissions; they're very different and the implementation effort differs significantly as well.

Separate issue because there is no reason to expect these would be worked on together/at the same time, and MAC policies are a lot more involved than using setuid or similar.

**Comment by liilac** [comment]
I think the project currently lacks expertise (I'm not an active part of the project anymore) in this area.

Happy to give guidance if there lands up being any interest in implementing this, though off-GitHub discussions have made it seem like there isn't a good understanding of the value of MAC.

TL;DR version is it gives much stronger protections and assurances around correct security-relevant behaviour; that traffic can't be routed to the clearnet, etc

Would suggest researching it a bit, as I believe it aligns with the project goals and adds value to the product.

@neuroscr
